src/clone-node.ts

```diff
@@ -192,35 +193,50 @@ function cloneScrollPosition<T extends HTMLElement>(
   }
 }
 
-function decorate<T extends HTMLElement>(nativeNode: T, clonedNode: T): T {
-  if (isInstanceOfElement(clonedNode, Element)) {
-    cloneCSSStyle(nativeNode, clonedNode)
-    clonePseudoElements(nativeNode, clonedNode)
-    cloneInputValue(nativeNode, clonedNode)
-    cloneSelectValue(nativeNode, clonedNode)
-    cloneScrollPosition(nativeNode, clonedNode)
-  }
-
-  return clonedNode
-}
-
 /**
  * TODO: re-add and optimise ensureSVGSymbols, run it as a "filter" - once per <use> tag
  * https://github.com/bubkoo/html-to-image/issues/341
  * on the original commit - it runs `clone.querySelectorAll('use')` on each node and `document.querySelector(id)` for each tag found
  */
 
-export async function cloneNode<T extends HTMLElement>(
+const isTextNode = (node: Node): node is Text => node.nodeType === 3 // Node.TEXT_NODE
+const isElementNode = (node: Node): node is HTMLElement => node.nodeType === 1 // Node.ELEMENT_NODE
+
+export async function cloneNode<T extends Node>(
   node: T,
   options: Options,
   isRoot?: boolean,
-): Promise<T | null> {
-  if (!isRoot && options.filter && !options.filter(node)) {
-    return null
+): Promise<Node | null> {
+  if (isTextNode(node)) {
+    return document.createTextNode(node.data)
+  }
+
+  if (!isElementNode(node)) {
+    return node.cloneNode(false) as HTMLElement
+  }
+
+  if (isRoot) {
+    // keeping for future improvements
   }
+  const style = window.getComputedStyle(node)
```

Comment:
Isn't this very slow?

Reply:
it is.. on my list for the next improvements